tchannel-core/src/main/java/com/uber/tchannel/channels/SubPeer.java

```diff
@@ -59,7 +60,13 @@ public Peer getPeer() {
         return peerManager.getPeer(remoteAddress);
     }
 
-    public boolean updateScore() {
+    public boolean updateScore(OutRequest outRequest) {
+        if (outRequest.isUsedPeer(remoteAddress)) {
+            // if we have used this peer, it should be given a lower score
+            score = SCORE_UNCONNECTED - 0.1;
```

Comment:
This is for retry logic. If a peer is tried before, it will have lower score than even unconnected peers. It will be reused when all "good" peers are exhausted.